pkg/resources/database_grant.go

```diff
@@ -0,0 +1,111 @@
+package resources
+
+import (
+	"fmt"
+
+	"github.com/hashicorp/terraform/helper/schema"
+	"github.com/hashicorp/terraform/helper/validation"
+
+	"github.com/chanzuckerberg/terraform-provider-snowflake/pkg/snowflake"
+)
+
+var validDatabasePrivileges = []string{
+	"ALL", "CREATE SCHEMA", "IMPORTED PRIVILEGES", "MODIFY", "MONITOR",
+	"OWNERSHIP", "REFERENCE_USAGE", "USAGE",
+}
+
+var databaseGrantSchema = map[string]*schema.Schema{
+	"database_name": &schema.Schema{
+		Type:        schema.TypeString,
+		Required:    true,
+		Description: "The name of the database on which to grant privileges.",
+		ForceNew:    true,
+	},
+	"privilege": &schema.Schema{
+		Type:         schema.TypeString,
+		Optional:     true,
+		Description:  "The privilege to grant on the database.",
+		Default:      "USAGE",
+		ValidateFunc: validation.StringInSlice(validDatabasePrivileges, true),
+		ForceNew:     true,
+	},
+	"roles": &schema.Schema{
+		Type:        schema.TypeSet,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Grants privilege to these roles.",
+		ForceNew:    true,
+	},
+	"shares": &schema.Schema{
+		Type:        schema.TypeSet,
+		Elem:        &schema.Schema{Type: schema.TypeString},
+		Optional:    true,
+		Description: "Grants privilege to these shares.",
+		ForceNew:    true,
+	},
+}
+
+// DatabaseGrant returns a pointer to the resource representing a database grant
+func DatabaseGrant() *schema.Resource {
+	return &schema.Resource{
+		Create: CreateDatabaseGrant,
+		Read:   ReadDatabaseGrant,
+		Delete: DeleteDatabaseGrant,
+
+		Schema: databaseGrantSchema,
+		Importer: &schema.ResourceImporter{
+			State: schema.ImportStatePassthrough,
+		},
+	}
+}
+
+// CreateDatabaseGrant implements schema.CreateFunc
+func CreateDatabaseGrant(data *schema.ResourceData, meta interface{}) error {
+	dbName := data.Get("database_name").(string)
+	builder := snowflake.DatabaseGrant(dbName)
+	priv := data.Get("privilege").(string)
+
+	err := createGenericGrant(data, meta, builder)
+	if err != nil {
+		return err
+	}
+
+	// ID format is <db_name>|||<privilege>
```

Comment:
It's kind of similar to Route53 records in AWS (https://github.com/Viostream/terraform-provider-aws/blob/master/aws/resource_aws_route53_record.go#L370) - they use the underscore as a separator which works because no part of the identifier is allowed to contain one.

Reply:
Ok, just to be clear I am curious about whether the combination of db_name and privilege is uniq across a snowflake account.